src/ApiService/ApiService/Functions/ReproVmss.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System.Net;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+
+namespace Microsoft.OneFuzz.Service.Functions;
+
+public class ReproVmss {
+    private readonly ILogTracer _log;
+    private readonly IEndpointAuthorization _auth;
+    private readonly IOnefuzzContext _context;
+
+    public ReproVmss(ILogTracer log, IEndpointAuthorization auth, IOnefuzzContext context) {
+        _log = log;
+        _auth = auth;
+        _context = context;
+    }
+
+    [Function("repro_vmss")]
+    public Async.Task<HttpResponseData> Run([HttpTrigger(AuthorizationLevel.Anonymous, "GET", "PATCH", "POST", "DELETE")] HttpRequestData req) {
+        return _auth.CallIfUser(req, r => r.Method switch {
+            "GET" => Get(r),
+            "POST" => Post(r),
+            "DELETE" => Delete(r),
+            _ => throw new InvalidOperationException("Unsupported HTTP method"),
+        });
+    }
+
+    private async Async.Task<HttpResponseData> Get(HttpRequestData req) {
+        var request = await RequestHandling.ParseRequest<ReproGet>(req);
+        if (!request.IsOk) {
+            return await _context.RequestHandling.NotOk(req, request.ErrorV, "repro_vm get");
+        }
+
+        if (request.OkV.VmId != null) {
+            var vm = await _context.ReproOperations.SearchByPartitionKeys(new[] { $"{request.OkV.VmId}" }).FirstOrDefaultAsync();
+
+            if (vm == null) {
+                return await _context.RequestHandling.NotOk(req, new Error(ErrorCode.INVALID_REQUEST, new[] { "no such VM" }), $"{request.OkV.VmId}");
+            }
+
+            var response = req.CreateResponse(HttpStatusCode.OK);
+            await response.WriteAsJsonAsync(vm);
+            return response;
+        }
+
+        var vms = await _context.ReproOperations.SearchStates(VmStateHelper.Available).Select(vm => vm with { Auth = null }).ToListAsync();
```

Comment:
Out of scope for this PR, but what do you think of creating an attribute called `Sensitive` (or similar) that clears out the information for that field when serializing? That way we wouldn't need to remember to do `vm with { Auth = null })` every time.

Reply:
Or I guess we could just change the `Authentication` record to 
```csharp
public record Authentication
(
    SecretData<string> Password,
    string PublicKey,
    SecretData<string> PrivateKey
);
```

Reply:
agreed. SecretData is the way to got for this. we have a work item (in ado) related to this. i plan on doing this after the migration.